src/core/thread/mle.cpp

```diff
@@ -4741,5 +4764,56 @@ void Mle::DelayedResponseMetadata::RemoveFrom(Message &aMessage) const
     SuccessOrAssert(aMessage.SetLength(aMessage.GetLength() - sizeof(*this)));
 }
 
+Error Mle::DetachGracefully(otDetachGracefullyCallback aCallback, void *aContext)
+{
+    Error error = kErrorNone;
+
+    OT_ASSERT(IsChild());
+
+    if (mDetachGracefullyTimer.IsRunning())
```

Comment:
It's a little tricky, but it's also detaching if `mDetachGracefullyTimer` is not running but the tasklet is posted.

Reply:
Yes. In that case the callback will be called immediately but the timer will still start. I moved the stopping of the timer into the tasklet handler so that stopping of the timer is always tied to calling of the callback.

Reply:
Tasklets are not called "immmediately". 

But moving stopping timer to tasklet should work. 
Thanks.